lib/services/android-project-service.ts

```diff
@@ -277,76 +232,13 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 
 	@performanceLog()
 	public async buildProject(projectRoot: string, projectData: IProjectData, buildConfig: IBuildConfig): Promise<void> {
-		let task;
-		const gradleArgs = this.getGradleBuildOptions(buildConfig, projectData);
-		const baseTask = buildConfig.androidBundle ? "bundle" : "assemble";
 		const platformData = this.getPlatformData(projectData);
-		const outputPath = buildConfig.androidBundle ? platformData.bundleBuildOutputPath : platformData.getBuildOutputPath(buildConfig);
-		if (this.$logger.getLevel() === "TRACE") {
-			gradleArgs.unshift("--stacktrace");
-			gradleArgs.unshift("--debug");
-		}
-		if (buildConfig.release) {
-			task = `${baseTask}Release`;
-		} else {
-			task = `${baseTask}Debug`;
-		}
-
-		gradleArgs.unshift(task);
-
-		const handler = (data: any) => {
-			this.emit(constants.BUILD_OUTPUT_EVENT_NAME, data);
-		};
-
-		await attachAwaitDetach(constants.BUILD_OUTPUT_EVENT_NAME,
-			this.$childProcess,
-			handler,
-			this.executeCommand({
-				projectRoot: this.getPlatformData(projectData).projectRoot,
-				gradleArgs,
-				childProcessOpts: { stdio: buildConfig.buildOutputStdio || "inherit" },
-				spawnFromEventOptions: { emitOptions: { eventName: constants.BUILD_OUTPUT_EVENT_NAME }, throwError: true },
-				message: "Gradle build..."
-			})
-		);
+		await this.$gradleBuildService.buildProject(platformData.projectRoot, buildConfig);
 
+		const outputPath = buildConfig.androidBundle ? platformData.bundleBuildOutputPath : platformData.getBuildOutputPath(buildConfig);
 		await this.$filesHashService.saveHashesForProject(this._platformData, outputPath);
 	}
 
-	private getGradleBuildOptions(settings: IAndroidBuildOptionsSettings, projectData: IProjectData): Array<string> {
-		const configurationFilePath = this.getPlatformData(projectData).configurationFilePath;
-
-		const buildOptions: Array<string> = this.getBuildOptions(configurationFilePath);
-
-		if (settings.release) {
-			buildOptions.push("-Prelease");
-			buildOptions.push(`-PksPath=${path.resolve(settings.keyStorePath)}`);
-			buildOptions.push(`-Palias=${settings.keyStoreAlias}`);
-			buildOptions.push(`-Ppassword=${settings.keyStoreAliasPassword}`);
-			buildOptions.push(`-PksPassword=${settings.keyStorePassword}`);
-		}
-
-		return buildOptions;
-	}
-
-	private getBuildOptions(configurationFilePath?: string): Array<string> {
-		this.$androidToolsInfo.validateInfo({ showWarningsAsErrors: true, validateTargetSdk: true });
```

Comment:
Is this validation obsolete?

Reply:
Yes, it is executed before the execution of each command, so no need to validate it here again.